src/mail.sh

```diff
@@ -340,16 +349,54 @@ function find_commit_references()
       fi
     done <<< "$parsed"
     parsed=''
+
   done <<< "$(git rev-parse -- $args 2> /dev/null)"
 
-  if [[ -n "$commit_range" ]]; then
+  if ! validate_patch_dir_reference "$patch_dir_references"; then
```

Comment:
Move this function call from outside the if, and just check `?`. Finally, use the `[[ ]]` pattern.